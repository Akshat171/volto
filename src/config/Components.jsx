import { PreviewImage } from '@plone/volto/components';
import TeaserDefaultBody from '@plone/volto/components/manage/Blocks/Teaser/DefaultBody';

export const components = {
  PreviewImage: { component: PreviewImage },
  'Teaser|Default': { component: TeaserDefaultBody },
};
