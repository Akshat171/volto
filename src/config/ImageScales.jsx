export const imageScales = {
  huge: 1600,
  great: 1200,
  larger: 1000,
  large: 800,
  teaser: 600,
  preview: 400,
  mini: 200,
  thumb: 128,
  tile: 64,
  icon: 32,
};
