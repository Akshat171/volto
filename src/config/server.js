import imagesMiddleware from '@plone/volto/express-middleware/images';
import filesMiddleware from '@plone/volto/express-middleware/files';
import robotstxtMiddleware from '@plone/volto/express-middleware/robotstxt';
import sitemapMiddleware from '@plone/volto/express-middleware/sitemap';
import devProxyMiddleware from '@plone/volto/express-middleware/devproxy';
import prefixPathMiddleware from '@plone/volto/express-middleware/prefixPath';

const settings = {
  expressMiddleware: [
    devProxyMiddleware(),
    prefixPathMiddleware(),
    filesMiddleware(),
    imagesMiddleware(),
    robotstxtMiddleware(),
    sitemapMiddleware(),
  ],
  criticalCssPath: 'public/critical.css',
  readCriticalCss: null, // so it will be defaultReadCriticalCss
  extractScripts: { errorPages: false },
};

export default settings;
