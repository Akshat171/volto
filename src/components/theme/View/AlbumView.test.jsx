import React from 'react';
import renderer from 'react-test-renderer';
import AlbumView from './AlbumView';

import config from '@plone/volto/registry';

config.settings.imageScales = {
  large: 768,
  preview: 400,
  mini: 200,
  thumb: 128,
  tile: 64,
  icon: 32,
  listing: 16,
};

test('renders an gallery view component', () => {
  const component = renderer.create(
    <AlbumView
      content={{
        title: 'Hello World!',
        description: 'Hi',
        items: [
          {
            '@id': 'http://localhost:3000/news',
            url: '/news',
            image: {
              download: 'file:///preview.jpg',
              scales: {
                preview: {
                  download: 'file:///preview.jpg',
                },
                large: {
                  download: 'file:///preview.jpg',
                },
              },
            },
            image_field: 'image',
          },
          {
            '@id': 'http://localhost:3000/events',
            url: '/events',
            image: {
              download: 'file:///preview.jpg',
              scales: {
                preview: {
                  download: 'file:///preview.jpg',
                },
                large: {
                  download: 'file:///preview.jpg',
                },
              },
            },
            image_field: 'image',
          },
        ],
      }}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
