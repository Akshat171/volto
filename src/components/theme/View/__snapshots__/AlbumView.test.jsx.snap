// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an gallery view component 1`] = `
<div
  className="ui container view-wrapper"
>
  <article
    id="content"
  >
    <header>
      <h1
        className="documentFirstHeading"
      >
        Hello World!
      </h1>
      <p
        className="documentDescription"
      >
        Hi
      </p>
    </header>
    <section
      id="content-core"
    >
      <div
        className="ui doubling stackable four column grid"
      >
        <div
          className="column"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="ui segment imageborder"
          >
            <picture
              className="volto-image"
            >
              <img
                alt=""
                loading="lazy"
                onClick={[Function]}
                role="img"
                src="http://localhost:3000/news"
                style={
                  Object {
                    "objectFit": "cover",
                    "width": "100%",
                  }
                }
                verticalAlign="middle"
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
              <img
                src=\\"http://localhost:3000/news\\"
                undefined
                alt=\\"\\"
                class=\\"\\"
                role=\\"img\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
                }
              }
            />
          </div>
        </div>
        <div
          className="column"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="ui segment imageborder"
          >
            <picture
              className="volto-image"
            >
              <img
                alt=""
                loading="lazy"
                onClick={[Function]}
                role="img"
                src="http://localhost:3000/events"
                style={
                  Object {
                    "objectFit": "cover",
                    "width": "100%",
                  }
                }
                verticalAlign="middle"
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
              <img
                src=\\"http://localhost:3000/events\\"
                undefined
                alt=\\"\\"
                class=\\"\\"
                role=\\"img\\"
                loading=\\"lazy\\"
                style=\\"width: 100%; object-fit: cover;\\"
            ",
                }
              }
            />
          </div>
        </div>
      </div>
    </section>
  </article>
</div>
`;
